xarray/tests/test_plot.py

```diff
@@ -624,6 +624,13 @@ def test_plot_nans(self):
         clim2 = self.plotfunc(x2).get_clim()
         self.assertEqual(clim1, clim2)
 
+    def test_can_plot_all_nans(self):
+        # regression test for issue #1780
+        DataArray(np.full((2, 2), np.nan)).plot.imshow()
```

Comment:
Do this instead:

```python
self.plotfunc(DataArray(np.full((2, 2), np.nan)))
```

`Common2dMixin` will check that all 2d functions work consistently